Libraries/LibWeb/DOM/EventTarget.cpp

```diff
@@ -423,12 +422,11 @@ WebIDL::CallbackType* EventTarget::get_current_value_of_event_handler(FlyString
 
         // 5. If element is not null and element has a form owner, let form owner be that form owner. Otherwise, let form owner be null.
         GC::Ptr<HTML::HTMLFormElement> form_owner;
-        if (is<HTML::FormAssociatedElement>(element.ptr())) {
-            auto* form_associated_element = dynamic_cast<HTML::FormAssociatedElement*>(element.ptr());
-            VERIFY(form_associated_element);
-
-            if (form_associated_element->form())
-                form_owner = form_associated_element->form();
+        if (element) {
```

Comment:
Do we need the outer `if`?
```c++
if (auto* form_associated_element = as_if<HTML::FormAssociatedElement>(element.ptr())) {
    if (auto* form = form_associated_element->form())
        form_owner = form;
}
```

Reply:
(on that note, I've also been wanting to add `as` overloads for `GC::Ptr` and friends so we don't have to add `.ptr()` in places like this)

Reply:
Oh for sure, I wasn't trying to suggest we do that here :)